internal/interface/repository/postgres/articles.go

```diff
@@ -0,0 +1,147 @@
+package postgres
+
+import (
+	"database/sql"
+	"fmt"
+	"github.com/mp-hl-2021/unarXiv/internal/domain"
+	"github.com/mp-hl-2021/unarXiv/internal/domain/model"
+	// "regexp"
+	// "strings"
+
+	_ "github.com/lib/pq"
+)
+
+type ArticleRepo struct {
+	db *sql.DB
+}
+
+func NewArticleRepo(db *sql.DB) *ArticleRepo {
+	return &ArticleRepo{db: db}
+}
+
+func (a *ArticleRepo) ArticleById(id model.ArticleId) (model.Article, error) {
+	rows, err := a.db.Query("SELECT Id, Title, Abstract, LastUpdateTimestamp FROM Articles WHERE Id = $1;", id)
+	if err != nil {
+		return model.Article{}, err
+	}
+	defer rows.Close()
+	for rows.Next() {
```

Comment:
This needs refactoring, in my view. Consider using fewer SQL queries to extract Article -> Authors data. Keep in mind the high cognitive load of deeply nested code.